.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,62 @@
+name: Continuous Integration
+on: 
+    push:
+        paths-ignore:
+            - 'README.md'
+    pull_request:
+        # Sequence of patterns matched against refs/heads
+        branches:    
+            - main  
+
+jobs:
+    job-one:
+        name: Build and test
+        runs-on: ubuntu-latest
+        steps:
+        - uses: actions/checkout@v2
+        - run: docker build --target test --tag todo-app:test .
+        - run: docker run todo-app:test
+    job-two:
+        name: Job Two
+        runs-on: ubuntu-latest
+        needs: job-one 
+        #if: ${{ github.event_name == 'pull_request' && github.ref == 'refs/heads/main'}}
+        steps:
+        - uses: actions/checkout@v2
+        - run: echo "Publishing!"
+        -
+          name: Login to Docker Hub
+          uses: docker/login-action@v2
+          with:
+            username: ${{ secrets.DOCKER_USERNAME }}
+            password: ${{ secrets.DOCKER_PASSWORD }}
+            
+        -
+          name: Build and push
+          uses: docker/build-push-action@v3
+          with:
+            push: true
+            tags: lhcapita1/todo-app
+            target: production
+        #- run: docker build --target production --tag lhcapita1/cc-todo-app .
+        #- run: docker push lhcapita1/cc-todo-app
+        #- run: docker tag lhcapita1/cc-todo-app registry.heroku.com/mod-8-todo-app/web
+        #- uses: CDNievas/heroku-action@v1.0 # This is the action
+        #  with:
+        #    heroku_email: ${{secrets.HEROKU_EMAIL}}
+        #    heroku_api_key: ${{secrets.HEROKU_API_KEY}}
+        #    heroku_app_name: "mod-8-todo-app"
+        - name: Build, Push and Deploy to Heroku
+          id: heroku
+          uses: Techmmunity/docker-heroku-deploy@v1.0.0 # use the latest version of the action
+          with:
+            email: ${{ secrets.HEROKU_EMAIL }} # your heroku email
+            api_key: ${{ secrets.HEROKU_API_KEY }} # your  heroku api key
+            app_name: "mod-8-todo-app" # you aplication name
+            options: "--target production" # OPTIONAL: Docker Build Options
```

Comment:
Does this get respected? I notice you've rearranged your Dockerfile so not sure if that's to avoid needing to specify this or not!